### PR TITLE
PDESolutions

### PR DESCRIPTION
**Pull request by xtalax**
Adds a solution wrapper type system for PDEs, and a solution wrapping function, to be extended in discretizer packages.

Must be merged BEFORE https://github.com/SciML/DiffEqBase.jl/pull/825

### CODE REVIEW

File: src/solutions/basic_solutions.jl [hunk 1]
```diff
@@ -77,3 +77,15 @@ function build_solution(prob::AbstractIntegralProblem,
                      typeof(prob), typeof(alg), typeof(chi)}(u, resid, prob, alg, retcode,
                                                              chi)
 end
+
+function wrap_sol(sol)
+    sys_type = if sol.prob.f.sys !== nothing
+        sol.prob.f.sys.metadata
```
Comment (ChrisRackauckas) [inline src/solutions/basic_solutions.jl:83]: shouldn't this just be done via `prob.problem_type` information?

Comment (xtalax) [reply src/solutions/basic_solutions.jl:83]: See https://github.com/SciML/DiffEqBase.jl/pull/825

Comment (ChrisRackauckas) [reply src/solutions/basic_solutions.jl:83]: why wait to change that?

Comment (xtalax) [reply src/solutions/basic_solutions.jl]: Done

**Pull request merged**